gems/aws-crt/ext/compile.rb

```diff
@@ -5,41 +5,55 @@
 require 'shellwords'
 require_relative '../lib/aws-crt/platforms'
 
-abort 'Missing cmake' unless find_executable 'cmake'
+CMAKE_PATH = find_executable('cmake3') || find_executable('cmake')
+abort 'Missing cmake' unless CMAKE_PATH
+CMAKE = File.basename(CMAKE_PATH)
 
 def cmake_version
-  version_str = `cmake --version`
+  version_str = `#{CMAKE} --version`
   match = /(\d+)\.(\d+)\.(\d+)/.match(version_str)
   [match[1].to_i, match[2].to_i, match[3].to_i]
 end
 
+CMAKE_VERSION = cmake_version
+
 # whether installed cmake supports --parallel build flag
 def cmake_has_parallel_flag?
-  (cmake_version <=> [3, 12]) >= 0
+  (CMAKE_VERSION <=> [3, 12]) >= 0
```

Comment:
Does this still work?

Reply:
yeah, I just stored the version in a global const a few lines up
since the `cmake_version()` function involved invoking an external program, it seemed like a good idea to just store it